### PR TITLE
Incorrect logic for handling responses due to changing teams #4484

### LINKED ISSUE #4484: Incorrect logic for handling responses due to changing teams
In FeedbackResponsesLogic's <code>updateFeedbackResponseForChangingTeam()</code>.
booleans

The conditions should include a check for FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF, otherwise questions with that recipient type will not be correctly handled. Related to #4482

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #4484

### CODE REVIEW

**Comment by damithc** [comment]
Any tests?

File: src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java [hunk 1]
```diff
@@ -51,6 +53,7 @@ public static void classSetUp() throws Exception {
         turnLoggingUp(FeedbackResponsesLogic.class);
         removeAndRestoreTypicalDataInDatastore();
         removeAndRestoreDatastoreFromJson("/SpecialCharacterTest.json");
+        removeAndRestoreDatastoreFromJson("/FeedbackSessionQuestionTypeTest.json");
```
Comment (LowWeiLin) [inline src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: Will using this json affect other tests using it?
If it won't, can merge this PR.

Comment (kanghj) [inline src/test/java/teammates/test/cases/logic/FeedbackResponsesLogicTest.java]: Updated to use new test data, to be safe.

**Comment by LowWeiLin** [comment]
Can merge!

**Pull request merged**